service.subtitles.subtitool/resources/lib/SubTiTool.py

```diff
@@ -0,0 +1,59 @@
+# *-* coding: utf-8 *-*
+
+import urllib, urllib2, base64, StringIO, sys, xbmc
+from xml.dom import minidom
+
+
+class SubTiToolHelper:
+    def __init__(self, filename, md5hash):
+        self.info = {}
+        self.filename = filename
+        self.url = "http://www.subtitool.com/api/"
+        self.md5hash = md5hash
+
+    def search(self, item, t, langs):
+
+        fulllang = xbmc.convertLanguage(item['preferredlanguage'], xbmc.ENGLISH_NAME)
+        if fulllang == "Persian": fulllang = "Farsi/Persian"
+        xbmc.executebuiltin("Notification(Title," + fulllang + ")")
+        url = "http://www.subtitool.com/api/?query=" + self.filename + "&Lang=" + langs
+        subs = urllib.urlopen(url).read()
+        DOMTree = minidom.parseString(subs)
+        if DOMTree.getElementsByTagName('Subtitle').length == 0:
+           try:
+            url = "http://www.subtitool.com/api/?query=" + self.filename + "&Lang=" + fulllang + "&OR=1"
+            subs = urllib.urlopen(url).read()
+            DOMTree = minidom.parseString(subs)
+           except Exception, e:
+                log("Subtitool","Not Found OR")
+
+           try:
+            url = "http://www.subtitool.com/api/?query=" + self.filename + "&Lang=" + fulllang
+            subs = urllib.urlopen(url).read()
+            DOMTree = minidom.parseString(subs)
+           except Exception, e:
+                log("Subtitool","Not Found")
+
+        return DOMTree
+
+    def download(self, dllink, language="PL"):
+
+        try:
+            response = urllib.urlopen(dllink)
+        except urllib2.URLError, e:
+            sys.stderr.write(e.message)
+            return False
+
+        try:
+            srtdata = response.read()
+            open(self.filename, "w").write(srtdata)
```

Comment:
oh, one more:
you should close the file again after writing, best by using with() statement.

with open(self.filename, "w") as file:
   file.write(srtdata)